src/cocaine-app/yt_worker.py

```diff
@@ -0,0 +1,176 @@
+        self.preaggregate_query = """
+$dateconv = @@
+import time
+import datetime
+def tounixdate(exp_time):
+    return int(time.mktime((datetime.date.fromtimestamp(exp_time) + datetime.timedelta(days=1)).timetuple()))
+@@;
+
+$dateconvert = Python::tounixdate("(Uint64?)->Uint64", $dateconv);
+
+INSERT INTO [{agr_table}]
+SELECT
+    couple_id,
+    expiration_date,
+    namespace,
+    "upload" AS operation,
+    SUM(CAST(object_size AS Int64)) AS expired_size,
+    "{date_iso}" AS source_table,
+    {timestamp} AS timestamp
+FROM [{main_table}]
+WHERE op="upload"
+GROUP BY
+    couple_id,
+    namespace,
+    $dateconvert(CAST(expire_at AS Uint64)) AS expiration_date;
+
+INSERT INTO [{agr_table}]
+SELECT
+    couple_id,
+    expiration_date,
+    namespace,
+    "delete" AS operation,
+    -1*SUM(CAST(object_size AS Int64)) AS expired_size,
+    "{date_iso}" AS source_table,
+    {timestamp} AS timestamp
+FROM [{main_table}]
+WHERE op="delete" AND expire_at IS NOT NULL
+GROUP BY
+    couple_id,
+    namespace,
+    $dateconvert(CAST(expire_at AS Uint64)) AS expiration_date;
+"""
+
+    def send_request(self, query, timeout=None):
+        """
+        Send request to YQL: ping verification + attempts retry
+        @return result dict on success, excepts otherwise
+        """
+        logger.debug('Send YQL request {}'.format(query))
+        result = None
+
+        with YqlClient(db=self._cluster, token=self._token) as yql:
+            try:
+                if not yql.ping():
+                    raise IOError("YQL ping failed {} {}".format(self._cluster, self._token))
+            except:
+                logger.exception("YQL ping excepted {}{}".format(self._cluster, self._token))
+                raise
+
+            start_time = time.time()
+
+            for attempt in xrange(self._attempts):
+                try:
+                    request = yql.query(query)
+                    request.run()
+
+                    logger.info("YQL query is running with id {}".format(request.operation_id))
+                    result = request.results
+                    end_time = time.time()
+                    if result.is_success:
+                        logger.info("YQL query succeeded and took {}".format(end_time-start_time))
+                        if timeout and (end_time - start_time > timeout):
+                            raise AssertionError("YQL query({}..{}) > timeout {}".format(start_time, end_time, timeout))
+                        return result
+
+                    logger.error("YQL failed #{}: {}[{}] ({},{!s})".format(attempt, result.status,
+                                 request.status_code, request.explain(), request.exc_info))
+                    for error in result.errors:
+                        logger.error("YQL query result error {!s}".format(error))
+
+                    time.sleep(self._delay)
+                except:
+                    logger.exception("YQL request excepted")
+                    raise
```

Comment:
Done in https://github.com/yandex/mastermind/pull/311/commits/4732106cd00d7b48fb9a6db30b7cff4aef8db6df